components/addressmanager/src/port_mapping_extender.rs

```diff
@@ -0,0 +1,82 @@
+use igd_next::{aio::tokio::Tokio, AddPortError};
+use kaspa_core::{
+    debug, error,
+    task::{
+        service::{AsyncService, AsyncServiceFuture},
+        tick::{TickReason, TickService},
+    },
+    trace,
+};
+use std::{net::SocketAddr, sync::Arc, time::Duration};
+
+pub const SERVICE_NAME: &str = "port-mapping-extender";
+
+pub struct Extender {
+    tick_service: Arc<TickService>,
+    fetch_interval: Duration,
+    deadline_sec: u64,
+    gateway: igd_next::aio::Gateway<Tokio>,
+    external_port: u16,
+    local_addr: SocketAddr,
+}
+
+impl Extender {
+    pub fn new(
+        tick_service: Arc<TickService>,
+        fetch_interval: Duration,
+        deadline_sec: u64,
+        gateway: igd_next::aio::Gateway<Tokio>,
+        external_port: u16,
+        local_addr: SocketAddr,
+    ) -> Self {
+        Self { tick_service, fetch_interval, deadline_sec, gateway, external_port, local_addr }
+    }
+}
+
+impl Extender {
+    pub async fn worker(&self) -> Result<(), AddPortError> {
+        while let TickReason::Wakeup = self.tick_service.tick(self.fetch_interval).await {
+            self.gateway
+                .add_port(
+                    igd_next::PortMappingProtocol::TCP,
+                    self.external_port,
+                    self.local_addr,
+                    self.deadline_sec as u32,
+                    "Kaspad-rusty",
+                )
+                .await
+                .unwrap(); // todo handle??
```

Comment:
Definitely needs to be handled. Currently can cause the node to panic.

Replication:
1. Start the node
2. Turn off your wifi